source-map.bs

```diff
@@ -394,6 +386,182 @@ When the [=Source Mapping URL=] is not absolute, then it is relative to the gene
+
+If a file is served through HTTP(S) with a `sourcemap` header, the value of the header is
+the URL of the linked source map.
+
+```
+sourcemap: <url>
+```
+
+Note: Previous revisions of this document recommended a header name of `x-sourcemap`.  This
+is now deprecated; `sourcemap` is now expected.
+
+### Linking through inline annotations
+
+The generated code should include a comment, or the equivalent construct depending on its
+language or format, named `sourceMappingURL` and that contains the URL of the source map. This
+specification defines how the comment should look like for JavaScript, CSS, and WebAssembly.
+Other languages should follow a similar convention.
+
+For a given language there can be multiple ways of detecting the `sourceMappingURL` comment,
+to allow for different implementations to choose what is less complex for them. The generated
+code <dfn>unambiguously links to a source map</dfn> if the result of all the extraction methods
+is the same.
+
+If a tool consumes one or more source files that [=unambiguously links to a source map=] and it
+produces an output file that links to a source map, it must do so [=unambiguously links to a
+source map|unambiguously=].
+
+<div class="example">
+The following JavaScript code links to a source map, but it does not do so [=unambiguously links
+to a source map|unambiguously=]:
+
+```js
+let a = `
+//# sourceMappingURL=foo.js.map
+//`;
+```
+
+Extracing a Source Map URL from it [=extract a Source Map URL from JavaScript through
+parsing|through parsing=] gives null, while [=extract a Source Map URL from JavaScript
+without parsing|without parsing=] gives `foo.js.map`.
+
+</div>
+
+#### Extraction methods for JavaScript sources
+
+To <dfn export>extract a Source Map URL from JavaScript through parsing</dfn> a [=string=] |source|,
+run the following steps:
+
+1. Let |tokens| be the [=list=] of [=tokens=]
+    obtained by parsing |source| according to [[ECMA-262]].
+1. [=For each=] |token| in |tokens|, in reverse order:
+    1. If |token| is not a [=single-line comment=] or a [=multi-line comment=], return null.
+    1. Let |comment| be the content of |token|.
+    1. If [=match a Source Map URL in a comment|matching a Source Map URL in=]
+        |comment| returns a [=string=], return it.
+
+To <dfn export>extract a Source Map URL from JavaScript without parsing</dfn> a [=string=] |source|,
+run the following steps:
+
+1. Let |lines| be the result of [=strictly split|strictly splitting=] |source| on [=line
+    terminator code points|ECMAScript line terminator code points=].
+1. Let |lastURL| be null.
+1. [=For each=] |line| in |lines|:
+    1. Let |position| be a [=position variable=] for |line|, initially pointing at the start of |line|.
+    1. [=While=] |position| doesn't point past the end of |line|:
+        1. [=Collect a sequence of code points=] that are [=white space code points|ECMAScript
+            white space code points=] from |line| given |position|.
```

Comment:
In the original PR there was this comment by @gibson042:

> Is it a problem that ECMAScript white space is subject to change over time as future Unicode editions change the set of code points in general category "Space_Separator"?

I think it's ok to expect implementations to evolve together with Unicode, but how do other folks feel?